### PR TITLE
EVG-3462: log error when no portrange is given

### PR DESCRIPTION
**Pull request by li-cindy**

### CODE REVIEW

File: cloud/docker_test.go [hunk 1]
```diff
@@ -112,6 +112,14 @@ func (s *DockerSuite) TestValidateSettings() {
...
 	}
 	s.Error(settingsInvalidPorts.Validate())
+
+	// error when missing port PortRange
+	settingsNoPortRange := &dockerSettings{
+		HostIP:     "127.0.0.1",
+		ImageID:    "docker_image",
+		ClientPort: 4243,
+	}
+	s.Error(settingsNoPortRange.Validate())
```
Comment (bsamek) [inline cloud/docker_test.go]: The original bug was a panic, so you should add a test that passes a nil port and then asserts `s.NotPanics`, probably in a new test.

File: cloud/docker.go [hunk 2]
```diff
@@ -68,13 +68,15 @@ func (settings *dockerSettings) Validate() error {
 		return errors.New("Port must not be blank")
 	}
 
-	if settings.PortRange != nil {
-		min := settings.PortRange.MinPort
-		max := settings.PortRange.MaxPort
+	if settings.PortRange == nil {
```
Comment (bsamek) [inline cloud/docker.go]: Go has a concept of a zero value, which for an integer is 0. If both `settings.PortRange.MinPort` and `settings.PortRange.MaxPort` are 0, then this will validate, but I suspect we won't be able to spawn a host. Should the check on line 78 actually be `<=` instead of `<`?

File: cloud/docker_test.go (hunk 1 above)
Comment (li-cindy) [reply cloud/docker_test.go]: done

File: cloud/docker.go (hunk 2 above)
Comment (li-cindy) [reply cloud/docker.go]: done
